ios/ios-share.h

```diff
@@ -0,0 +1,20 @@
+// SPDX-License-Identifier: GPL-2.0
+#ifndef IOSSHARE_H
+#define IOSSHARE_H
+
+// onlt Qt headers and data structures allowed here
+#include <QString>
+#include <QObject>
+
+class IosShare : QObject {
+public:
+	IosShare(QObject *parent);
+	~IosShare(void);
```

Comment:
Declaring functions with `(void)` works for compatibility reasons, but is actually very weird in C++. This comes from the bad old times of K&R C where you didn't declare full function signatures.`int f();` meant `f` is a function returning int, but you don't say which arguments it has, because that would let the compiler help you and we don't want that (or rather we couldn't do that in 32KB of RAM). So later they had to introduce `int f(void)` to explicitly say that the function takes no arguments. However, in C++ you *must* declare functions, because the compiler must know whether arguments are passed by value or by reference [e.g. `int f(int i)` vs. `int f(int &i)`]. Thus, `int f()` became the idiomatic (and logical!) way of declaring a function without arguments.

Reply:
cut and paste from an example 😇
fixed.